### ISSUE TITLE
ovStoreBucketizer - Segmentation default

### ISSUE THREAD

**Issue by nottwy** [marbl/canu#435]
# Main content:
Dear developers,

I meet this problem and can't find a solution now.

It occurs in stage 1: correction.
It occurs in substage: bucketize.

The total error log is in the supplementary file below.
[1-bucketize.5205446_8.txt](https://github.com/marbl/canu/files/887518/1-bucketize.5205446_8.txt)

Wait for your reply.

# Additional information:

I run ovStoreBucketizer individually with this commandline:
```
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer \
        -O /HOME/path/to/correction/xiyu.ovlStore.BUILDING \
        -G /HOME/path/to/correction/xiyu.gkpStore \
        -C /HOME/path/to/correction/xiyu.ovlStore.BUILDING/config \
        -job  1 \
        -i /HOME/path/to/correction/1-overlapper/results/000001.ovb
```
The error log is:

```
maxError fraction: 1.000 percent: 100.000 encoded: 4095
Bucketizing /HOME/path/to/correction/1-overlapper/results/000001.ovb

Failed with 'Segmentation fault'

Backtrace (mangled):

/HOME/fsgen_1/Biosoftware/canu/canu-1.4/Linux-amd64/bin/ovStoreBucketizer(_Z17AS_UTL_catchCrashiP7siginfoPv+0x27)[0x404b97]
/lib64/libpthread.so.0[0x324640f710]
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer[0x40af94]
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer[0x4027ed]
/lib64/libc.so.6(__libc_start_main+0xfd)[0x3245c1ed5d]
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer[0x402159]

Backtrace (demangled):

[0] /HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer::AS_UTL_catchCrash(int, siginfo*, void*) + 0x27  [0x404b97]
[1] /lib64/libpthread.so.0() [0x324640f710]
[2] /HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer() [0x40af94]
[3] /HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer() [0x4027ed]
[4] /lib64/libc.so.6::(null) + 0xfd  [0x3245c1ed5d]
[5] /HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer() [0x402159]

GDB:


resume_after_05_ovStoreBucketizer.sh: line 7:  6433 Segmentation fault      
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer \
-O /HOME/path/to/correction/xiyu.ovlStore.BUILDING \
-G /HOME/path/to/correction/xiyu.gkpStore \
-C /HOME/path/to/correction/xiyu.ovlStore.BUILDING/config \
-job 1 \
-i /HOME/path/to/correction/1-overlapper/results/000001.ovb
```

**Comment by skoren** [comment]
What version of Canu is this? If it is 1.4 it could be issue #319. You can try updating to the latest code and re-starting Canu.

**Comment by nottwy** [comment]
The version of canu I'm using is 1.4.
I just installed it one week ago.

**Comment by nottwy** [comment]
-rwxr-xr-x 1 fsgen_1 fsgen  14K Apr  1 17:44 1-bucketize.sh
-rw-r--r-- 1 fsgen_1 fsgen  338 Apr  1 17:22 1-bucketize.jobSubmit.sh
-rwxr-xr-x 1 fsgen_1 fsgen 1.9K Apr  1 17:04 2-sort.sh
-rwxr-xr-x 1 fsgen_1 fsgen  501 Apr  1 17:04 3-index.sh
-rwxr-xr-x 1 fsgen_1 fsgen 1.3K Apr  1 17:04 0-config.sh

Here is my understanding about this step:
1. the script 0-config.sh has finished;
2. then we run the script 1-bucketize.jobSubmit.sh. It will invoke 1-bucketize.sh;
3. In the script 1-bucketize.sh, it will create lots of ovStoreBucketizer tasks.

I can run ovStoreBucketizer individually and get the same error message. I run it like this:
```
rm -r /HOME/path/to/correction/xiyu.ovlStore.BUILDING/create0001
/HOME/path/to/canu-1.4/Linux-amd64/bin/ovStoreBucketizer \
        -O /HOME/path/to/correction/xiyu.ovlStore.BUILDING \
        -G /HOME/path/to/correction/xiyu.gkpStore \
        -C /HOME/path/to/correction/xiyu.ovlStore.BUILDING/config \
        -job  1 \
        -i /HOME/path/to/correction/1-overlapper/results/000001.ovb
```

Then I will get the error log which I present above.

**Comment by skoren** [comment]
Yes, your understanding is correct, the issue is a known bug in 1.4. If you update to the latest from the repository (rather than a release) it should be fixed. You could build the latest code and just replace all files named ```ovStore*``` in your 1.4 installation with the updated ones to fix the bug. Version 1.5 should be released within a couple of weeks if you want to wait for that instead.

**Comment by nottwy** [comment]
I tried to download from the repository and install canu.
I type: `make` under the directory `src` but it failed.
I also tried `make -j 1`. It failed, too.

Here is the error message:
```
Building for 'Linux' '2.6.32-431.TH.x86_64' as 'amd64' into '/HOME/path/to/canu-1.4-20170403/Linux-amd64/{bin,obj}'
CC cc CXX g++
Building release v1.4

cc -o /HOME/path/to/canu-1.4-20170403/Linux-amd64/obj/libcanu.a/AS_UTL/libbacktrace/dwarf.o -c -MD -I/HOME/path/to/canu-1.4-20170403/src -IAS_UTL -Istores -Istores/libsnappy -Ialignment -Iutgcns/libNDalign -Iutgcns/libcns -Iutgcns/libpbutgcns -Iutgcns/libNDFalcon -Iutgcns/libboost -Imeryl/libleaff -IoverlapInCore -IoverlapInCore/libedlib -IoverlapInCore/liboverlap -Ifalcon_sense/libfalcon AS_UTL/libbacktrace/dwarf.c
AS_UTL/libbacktrace/dwarf.c: In function ‘dwarf_lookup_pc’:
AS_UTL/libbacktrace/dwarf.c:2779: error: ‘__ATOMIC_ACQUIRE’ undeclared (first use in this function)
AS_UTL/libbacktrace/dwarf.c:2779: error: (Each undeclared identifier is reported only once
AS_UTL/libbacktrace/dwarf.c:2779: error: for each function it appears in.)
AS_UTL/libbacktrace/dwarf.c:2839: error: ‘__ATOMIC_RELEASE’ undeclared (first use in this function)
AS_UTL/libbacktrace/dwarf.c: In function ‘dwarf_fileline’:
AS_UTL/libbacktrace/dwarf.c:2974: error: ‘__ATOMIC_ACQUIRE’ undeclared (first use in this function)
AS_UTL/libbacktrace/dwarf.c: In function ‘backtrace_dwarf_add’:
AS_UTL/libbacktrace/dwarf.c:3107: error: ‘__ATOMIC_ACQUIRE’ undeclared (first use in this function)
make: *** [/HOME/path/to/canu-1.4-20170403/Linux-amd64/obj/libcanu.a/AS_UTL/libbacktrace/dwarf.o] Error 1

```

**Comment by brianwalenz** [comment]
Looks like it picked up a non-gnu (or very old gnu) compiler in the 'cc' command.

Try setting CC=gcc in your environment, or possibly 'make CC=gcc'.

**Comment by brianwalenz** [comment]
See also issue #438.  Summary: `make BUILDSTACKTRACE=0`.

**Comment by nottwy** [comment]
Don't worry about me now ; ).I have compiled gcc 6.3.0 and successfully installed the latest canu from repository. Iy seems like now everything goes well.

3ku brianwalenz and skoren.

**Comment by skoren** [comment]
Closing, since you said it was working for you now. Open another issue if you encounter other errors.
